File: Search in a Row-Column sorted matrix.java
/* Given a 2D matrix mat[][], where every row and column is sorted in increasing order, and a number x the task is to find whether element x is present in the matrix or not.
  */

class Solution {
    public static boolean matSearch(int mat[][], int x) {
        // your code here
        if(mat==null || mat.length==0||mat[0].length==0){
            return false;
        }
        for(int i=0;i< mat.length;i++){
            for(int j=0;j< mat[i].length;j++){
                if(mat[i][j]==x){
                    return true;
                }
            }
        }
        return false;
    }
}

File: Search in a Row-Column sorted matrix.py
'''Given a 2D matrix mat[][], where every row and column is sorted in increasing order, and a number x the task is to find whether element x is present in the matrix or not.'''

#User function Template for python3
class Solution:
    def matSearch(self,mat, x):
        check=False
        # Complete this function
        for i in range(0,len(matrix)):
            if matrix[i][0]<=x and matrix[i][-1]>=x:
                for j in range(0,len(matrix[i])):
                    if matrix[i][j]==x:
                        check=True
                        break
               
        return check
